tests/test_state.py

```diff
@@ -1656,6 +1656,9 @@ async def background_task(self):
                 pass  # update proxy instance
 
         async with self:
+            # Methods on ImmutableMutableProxy should return their wrapped return value.
+            assert self.dict_list.pop("foo") is not None
```

Comment:
Should we check the actual value of `foo` rather than `not None` ?

Reply:
might not hurt. i'll add it to a subsequent PR